prov/psm2/src/psmx2.h

```diff
@@ -651,8 +651,8 @@ struct psmx2_fid_cntr {
 	struct psmx2_fid_domain	*domain;
 	int			events;
 	uint64_t		flags;
-	atomic_t		counter;
-	atomic_t		error_counter;
+	atomic32_t		counter;
+	atomic32_t		error_counter;
```

Comment:
Thanks for proposed enhancement. I'm going to add it tomorrow and update both PSMs in an appropriate way.